main.cpp

```diff
@@ -29,25 +29,56 @@ const IR::P4Program *prune_statements(const IR::P4Program *program,
                                       int required_exit_code) {
     cstring stripped_name = remove_extension(options.file);
     stripped_name += "_stripped.p4";
-    for (int j = 0; j < 20; j++) {
-        P4PRUNER::Pruner *pruner = new P4PRUNER::Pruner();
-
+    int same_before_pruning = 0;
+    int max_statements = 100;
+    for (int i = 0; i < 50; i++) {
```

Comment:
This code here is starting to get a bit long. There are two major segments (collecting and pruning) so we can break this into two functions.